sol/main.c

```diff
@@ -664,20 +664,21 @@ run_sol(struct net_config *net_conf, struct sol_config *sol_conf)
  * There should be only one sol_config instance.
  * Return an error if trying to allocate the second instance.
  *
- * Use rte_calloc() to zero-out the instance and initialize the
+ * Use rte_calloc_socket() to zero-out the instance and initialize the
  * request queue list to guarantee that cleanup_sol() won't fail
  * during initialization.
  */
 struct sol_config *
-alloc_sol_conf(void)
+alloc_sol_conf(unsigned int lcore)
```

Comment:
Given that `alloc_sol_conf()` now receives the lcore, set field `lcore_id` before returning and drop line `sol_conf.lcore_id = lcore` from `lua/sol.lua`.